docs/4.0-migration-guide/platform-entropy-gathering.md

```diff
@@ -0,0 +1,63 @@
+## Platform-specific Entropy Gathering
+
+### Disclaimer
+
+All the following sections refer to the case where entropy module uses default
+entropy sources, i.e. `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES` is not defined.
+
+### Background
+
+Up to Mbed TLS 3.6 entropy module supported 3 entropy sources by default:
+
+- Platform
+  - Enabled by default it could be disabled through `MBEDTLS_NO_PLATFORM_ENTROPY`.
+  - It used platform-specific sources such as getrandom(), /dev/urandom or
+    BCryptGenRandom() to gather entropy data.
+  - Only useful when running on high level OSes, not in baremetal projects.
+
+- Hardware
+  - Disabled by default, it could be enabled through `MBEDTLS_ENTROPY_HARDWARE_ALT`.
+  - It allowed the user to define a custom function named `mbedtls_hardware_poll()`
+    to gather entropy data from some hardware source.
+  - Very useful for baremetal projects.
+  - Unfortunately the prototype for the function was in a private header file.
+
+- NV seed
+  - Disabled by default, it could be enabled through `MBEDTLS_ENTROPY_NV_SEED`.
+  - It allowed the user to store an entropy seed on a non-volatile (NV) memory
+    (ex: filesystem).
+  - This is crucial on systems that do not have a cryptographic entropy source
+    available through the previous 2 options.
+
+Up to Mbed TLS 3.6 all these 3 build symbols could be enabled independently.
+
+### Reasons to change
```

Comment:
I don't think a detailed list of reasons is useful in the migration guide. That belongs in an architecture document. The migration guide is a practical document for programmers using the library.